core/src/main/scala/io/qbeast/core/transform/LinearTransformer.scala

```diff
@@ -58,20 +58,17 @@ case class LinearTransformer(columnName: String, dataType: QDataType) extends Tr
   override def makeTransformation(row: String => Any): Transformation = {
     val minAux = row(colMin)
     val maxAux = row(colMax)
-    if (minAux == null && maxAux == null) {
-      // If all values are null,
-      // we return a Transformation where null values are transformed to 0
-      NullToZeroTransformation
-    } else if (minAux == maxAux) {
-      // If both values are equal we return an IdentityTransformation
-      IdentityToZeroTransformation(minAux)
-    } else { // otherwise we pick the min and max
+    if (minAux == maxAux) {
+      val identityValue = getValue(minAux)
+      dataType match {
```

Comment:
And manually throw an error if it is not a `OrderedDataType`?

Reply:
it felt cleaner, but is true that it can accommodate other types if needed. 

I would suggest to manually throw an error when is not a OrderedDataType anyways. It would be simpler for debugging and understand what happened.